cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,52 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(vaps), pluralize.Pluralize(len(policies)+len(vaps), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	policyToUnknownRules := make(map[string]string)
+	var policyerr string
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var policyExist bool
+		ruleExists := false
+		for _, policy := range policies {
+			if policy.IsNamespaced() {
+				ns = policy.GetNamespace()
+			}
+			// ensure that the policy specified in the results matches one of the policies provided in the given files.
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() || res.Policy == ns+"/"+policy.GetName() {
+				policyExist = true
+
+				// ensure that the rule specified in the results exists in the policy.
+				for _, rule := range autogen.ComputeRules(policy) {
```

Comment:
I think before doing so, we could check the policy type and if it is of type `ValidatingAdmissionPolicy`, we could skip checking the rule name. WDYT?